cmd/istio_ca/main.go

```diff
@@ -121,13 +127,22 @@ func runCA() {
 		if value, exists := os.LookupEnv(namespaceKey); exists {
 			opts.namespace = value
 		}
+
+		if opts.namespace == "" {
+			glog.Fatal("Fail to get the namespace of this CA")
+		}
```

Comment:
This will not work with the current templates in istio/istio, because for cluster-wide, namespace is passed as "".
It is too late to make this change in 0.2

Reply:
Suggestion is to make the namespace flags changes in 0.3, since it needs to cascade all the way to pilot and istio, and requires change in the yaml files.

The actual certificate changes lgtm.